packages/neon_framework/lib/src/storage/sqlite_cookie_persistence.dart

```diff
@@ -0,0 +1,376 @@
+import 'package:sqflite/sqflite.dart';
+import 'package:universal_io/io.dart' show Cookie;
+
+const String _unsupportedCookieManagementMessage =
+    'The neon project does not plan to support individual cookie management.';
+
+/// An SQLite backed cookie persistence.
+///
+/// No maximum cookie age is set as mentioned in [RFC6265 section 7.3](https://httpwg.org/specs/rfc6265.html#rfc.section.7.3).
+///
+/// This persistence does not support cookie management through the [deleteAll]
+/// and [deleteWhere] methods. Calling them will throw a [UnsupportedError].
+@internal
+final class SQLiteCookiePersistence implements CookiePersistence {
+  /// Creates a new SQLite backed cookie persistence for the given account.
+  ///
+  /// Optionally [allowedBaseUri] can be used to restrict storage and loading of cookies to a certain domain and path.
+  SQLiteCookiePersistence({
+    required this.accountID,
+    Uri? allowedBaseUri,
+  }) : allowedBaseUri = switch (allowedBaseUri) {
+          null => null,
+          Uri(:final path) when path.endsWith('/') => Uri(
+              host: allowedBaseUri.host,
+              path: path.substring(0, path.length - 1),
+            ),
+          _ => allowedBaseUri,
+        };
+
+  final _log = Logger('SQLiteCookiePersistence');
+
+  /// The allowed cookie base uri.
+  ///
+  /// When not null, only cookies  domain and path matching this uri will be
+  /// persisted.
+  ///
+  /// See:
+  ///   * domain matching [RFC6265 section 5.1.3](https://httpwg.org/specs/rfc6265.html#rfc.section.5.1.3)
+  ///   * path matching [RFC6265 section 5.1.4](https://httpwg.org/specs/rfc6265.html#rfc.section.5.1.4)
+  final Uri? allowedBaseUri;
+
+  /// The id of the requesting account.
+  final String accountID;
+
+  @visibleForTesting
+  static Database? database;
+
+  /// Initializes this cookie persistence by setting up the backing SQLite database.
+  ///
+  /// This must called and completed before accessing other methods of the cache.
+  static Future<void> init() async {
+    if (database != null) {
+      return;
+    }
+
+    assert(
+      NeonPlatform.instance.canUsePaths,
+      'Tried to initialize SQLiteCookiePersistence on a platform without support for paths',
+    );
+    final cacheDir = await getApplicationCacheDirectory();
+    database = await openDatabase(
+      p.join(cacheDir.path, 'cookies.db'),
+      version: 1,
+      onCreate: onCreate,
+    );
+  }
+
+  @visibleForTesting
+  static Future<void> onCreate(Database db, int version) async {
+    // https://httpwg.org/specs/rfc6265.html#storage-model with an extra account key.
+    await db.execute('''
+CREATE TABLE "cookies" (
+  "account"           TEXT NOT NULL,
+  "name"              TEXT NOT NULL,
+  "value"             TEXT NOT NULL,
+  "expiry-time"       INTEGER NOT NULL,
+  "domain"            TEXT NOT NULL,
+  "path"              TEXT NOT NULL,
+  "creation-time"     INTEGER NOT NULL,
+  "last-access-time"  INTEGER NOT NULL,
```

Comment:
We only need it if we want to evic cookies (setting a max amount of cookies to store globally or per domain):
> If two cookies have the same removal priority, the user agent MUST evict the cookie with the earliest last-access date first.

We should consider doing so although I'm not sure what values to choose.
OTOH we should also consider adding a maximum time a cookie is stored.
I'm open for doing it in this PR or as a followup if we need to discuss this further.

Reply:
I don't think we will ever have a limited number of cookies (and in reality there are only 5 or so that are actually used by nextcloud) so the last access time is only useful for evicting old cookies. I don't know what a good age would be, but probably something between 1 month and 1 year.

Reply:
In that case we can remove the access time and just clamp the max age to now+something

Reply:
no we still need it: cookies need to be deleted where `last access time < now - 1 year` (so last updated a year ago or longer). we need some reference to when they were last used.